users/repo.go

```diff
@@ -0,0 +1,42 @@
+package users
+
+import (
+	"budget-helper/database"
+	"budget-helper/models"
+	"context"
+	"log"
+)
+
+type UserRepo struct {
+	db  *database.Database
+	ctx context.Context
+}
+
+func NewUserRepo(db *database.Database) *UserRepo {
+	return &UserRepo{db, context.Background()}
+}
+
+func (u *UserRepo) GetAllUsers() ([]*User, error) {
+	res, err := models.Users().All(u.ctx, u.db)
+	if err != nil {
+		log.Fatalf("error: %v\n", err)
+	}
+
+	userList := []*User{}
```

Comment:
Is this needed? I'm guessing that sqlboiler does this automatically in `All`. What's the type of `res` (I can't build yet)?

Reply:
`res` returns a `models.UserSlice` which is an SQLBoiler-specific struct generated when running `sqlboiler sqlite3`. I am using my own `User` struct and creating if with values extracted from the model so that I can more easily define my own type with its own app-specific functions without having to rely on/extend the one created by SQLBoiler. 

I'm not sure if there is a real technical benefit to this method, but it makes it easier for me to manage and keep track of.

Reply:
This type is an alias over `[]*models.User`, you don't need to re-cast it.

In your users package, just import this and hang whatever functions you need off of it. Eg:

```
func (u models.User) String() string {
    return u.Username
}
```

No need to have duplicate `models/user` and `users/user`.

Reply:
So to clarify, you can return `(models.UserSlice, error)` directly, with `return res, nil`.